### PR TITLE
Convert shapes to pyglet's new type annotation + docstring style

### PR DESCRIPTION
**Pull request by pushfoo**
TL;DR: Help with #1084 by addressing shapes.

### Changes

1. Move typing from docstrings to signatures:
   * Use the pipe syntax since it's been approved and somewhat tested on 3.8 with `from __future__ import annotations`
   * Place some annotations on default class attributes, but not all
   * Add comments explaining why
2. Convert docstrings to new formats
3. Add improvements for clarity and usability:
   1. Fix outdated references to blend modes and opacity in `__init__` doc
   2. Document broken `ShapeBase.group` behavior when `self.batch == None`
   3. Attempt to clarify rotation and anchor behavior
   4. Add cross-references
   5. Use `.. warning::` and :.. tip::` directives
   6. Use more flexible typing on `Polygon` and `MultiLine` types since they seem to target advanced users

### CODE REVIEW

File: pyglet/shapes.py [hunk 1]
```diff
@@ -9,8 +9,22 @@
 The Python ``in`` operator can be used to check whether a point is inside a shape.
 (This is approximated with some shapes, such as Star).
 
-To create more complex shapes than what is provided here, the lower level
-graphics API is more appropriate. See the :ref:`guide_graphics` for more details.
+If you the shapes in this module don't suit your needs, you
```
Comment (benmoran56) [inline pyglet/shapes.py:12]: Looks like an extra "you" here.

File: pyglet/shapes.py [hunk 2]
```diff
@@ -345,128 +377,167 @@ def rotation(self, rotation: float) -> None:
...
         self._vertex_list.rotation[:] = (rotation,) * self._num_verts
 
-    def draw(self):
-        """Draw the shape at its current position.
+    def draw(self) -> None:
+        """Debug method to draw a single shape at its current position.
+
+        .. warning:: Avoid this inefficient method for everyday use!
+
+                     Instead, add shapes to a :py:class:`Batch` and call
```
Comment (benmoran56) [inline pyglet/shapes.py:385]: "Instead" is being used twice, which can be simplified.

**Review by benmoran56 — APPROVED** [review]
